Readability.js

```diff
@@ -1721,7 +1724,14 @@ Readability.prototype = {
 
     this.log("Grabbed: " + articleContent.innerHTML);
 
-    this._postProcessContent(articleContent);
+    var baseElementPath;
+    // Determine the path base if there is a "base" element
+    var baseElement = this._doc.getElementsByTagName('base');
+    if (baseElement.length > 0) {
+      baseElementPath = baseElement[0].getAttribute("href");
```

Comment:
As noted in my comment on the PR, I think we should use `this._doc.baseURI` - perhaps closer to the point where we need it (rather than passing it around).

Also, ideally we would use a `URL` constructor in the code that gets a `baseURI` rather than the manual URL munging. So something like:

```js
var baseElement = this._doc.getElementsByTagName('base');
if (baseElement.length) {
  baseElementPath = new URL(baseElement[0].getAttribute("href"), this._uri.pathBase).href;
}
```

For this we'd have to up the node requirement to at least 7.0, which I think should be fine. I'll also have some issues integrating it into Firefox, but I'll deal with that when we get there. :-)

Reply:
@gijsk, this definitely makes sense and I'll try to get to this tonight. To the bugs you reference, do you think there's a better way of doing this that wouldn't require a change to this library? Thanks for your feedback!

Reply:
I think fixing the library is the 'right' fix, that we probably should have pursued in the bugzilla entries if I had considered it at the time.

But yeah, if you need a workaround, the Readability constructor takes a URI as argument. Instead of passing URI information out of `document.location` as noted in the readme, you could pass it out of a URL constructed from `document.baseURI`.

In fact, perhaps we could remove the URI argument for the Readability constructor entirely with some work. But don't worry about that as part of this PR.

Reply:
(And thank you for the PR! It's great to see a steady stream of people finding this library useful and contributing back fixes to make it work better.)